fuel-vm/src/interpreter/alu/wideint.rs

```diff
@@ -0,0 +1,365 @@
+                    }
+
+                    *of = overflow as Word;
+                    *err = 0;
+
+                    write_bytes(&mut self.memory, owner_regs, dest_addr, wrapped.to_be_bytes())?;
+
+                    inc_pc(pc)
+                }
+
+                pub(crate) fn [<alu_wideint_div_ $t:lower>](
+                    &mut self,
+                    dest_addr: Word,
+                    b: Word,
+                    c: Word,
+                    args: DivArgs,
+                ) -> Result<(), RuntimeError> {
+                    let owner_regs = self.ownership_registers();
+                    let (SystemRegisters { flag, mut of, mut err, pc, .. }, _) = split_registers(&mut self.registers);
+
+                    // LHS is always indirect
+                    let lhs: $t = $t::from_be_bytes(read_bytes(&self.memory, b)?);
+
+                    // RHS is only indirect if the flag is set
+                    let rhs: $t = if args.indirect_rhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, c)?)
+                    } else {
+                        c.into()
+                    };
+
+                    let lhs = [<to_prim_ $t:lower>](lhs);
+                    let rhs = [<to_prim_ $t:lower>](rhs);
+
+                    let result = match lhs.checked_div(rhs) {
+                        Some(d) => [<from_prim_ $t:lower>](d),
+                        None => {
+                            if is_unsafe_math(flag.into()) {
+                                $t::default() // Zero
+                            } else {
+                                return Err(PanicReason::ErrorFlag.into());
+                            }
+                        }
+                    };
+
+                    *of = 0;
+                    *err = 0;
+
+                    write_bytes(&mut self.memory, owner_regs, dest_addr, result.to_be_bytes())?;
+
+                    inc_pc(pc)
+                }
+
+                pub(crate) fn [<alu_wideint_addmod_ $t:lower>](
+                    &mut self,
+                    dest_addr: Word,
+                    b: Word,
+                    c: Word,
+                    d: Word,
+                ) -> Result<(), RuntimeError> {
+                    let owner_regs = self.ownership_registers();
+                    let (SystemRegisters { flag, mut of, mut err, pc, .. }, _) = split_registers(&mut self.registers);
+
+                    let lhs: $t = $t::from_be_bytes(read_bytes(&self.memory, b)?);
+                    let rhs: $t = $t::from_be_bytes(read_bytes(&self.memory, c)?);
+                    let modulus: $t = $t::from_be_bytes(read_bytes(&self.memory, d)?);
+
+                    let result: $t = if modulus == 0 {
+                        if is_unsafe_math(flag.into()) {
+                            $t::default() // Zero
+                        } else {
+                            return Err(PanicReason::ErrorFlag.into());
```

Comment:
Should probably use a more descriptive panic reason other than `ErrorFlag`.

Reply:
Non-wideint ALU operations use ErrorFlag for any error condition, including divion by zero. I should probably update them as well, if we're going to that direction.

Reply:
Done, the new flag is named `ArithmeticError`